docs/contributor/06-module-migration-guide.md

```diff
@@ -0,0 +1,231 @@
+      - channel `dev` => `1.35.0-rc1`
+  - `/stage`
+    - `module-release-meta.yaml` with
+      - channel `regular` => `1.32.0`
+      - channel `fast` => `1.34.0`
+      - channel `experimental` => `1.34.0-experimental`
+  - `/prod`
+    - `module-release-meta.yaml` with
+      - channel `regular` => `1.32.0`
+      - channel `fast` => `1.34.0`
+- `/kustomizations`
+  - `/dev`
+    - `/kustomization.yaml` with
+      - ModuleReleaseMeta `../../telemetry/dev/module-release-meta.yaml`
+      - ModuleTemplate `../../telemetry/moduletemplate-telemetry-1.32.0.yaml`
+      - ModuleTemplate `../../telemetry/moduletemplate-telemetry-1.34.0.yaml`
+      - ModuleTemplate `../../telemetry/moduletemplate-telemetry-1.34.0-experimental.yaml`
+      - ModuleTemplate `../../telemetry/moduletemplate-telemetry-1.35.0-rc1.yaml`
+      - ... (resources from other modules)
+  - `/stage`
+    - `/kustomization.yaml` with
+      - ModuleReleaseMeta `../../telemetry/stage/module-release-meta.yaml`
+      - ModuleTemplate `../../telemetry/moduletemplate-telemetry-1.32.0.yaml`
+      - ModuleTemplate `../../telemetry/moduletemplate-telemetry-1.34.0.yaml`
+      - ModuleTemplate `../../telemetry/moduletemplate-telemetry-1.34.0-experimental.yaml`
+      - ... (resources from other modules)
+  - `/prod`
+    - `/kustomization.yaml` with
+      - ModuleReleaseMeta `../../telemetry/stage/module-release-meta.yaml`
+      - ModuleTemplate `../../telemetry/moduletemplate-telemetry-1.32.0.yaml`
+      - ModuleTemplate `../../telemetry/moduletemplate-telemetry-1.34.0.yaml`
+      - ... (resources from other modules)
+
+### 3) Verify in KCP
+
+ArgoCD picks up and deploys the changes from step 2). All landscapes have the same channel-version mapping of the module described in OLD and NEW metadata.
+
+Following the example above, the following resources exist in KCP:
+
+#### New ModuleTemplates
+- ModuleTemplate `kyma-system/telemetry-regular` pointing to `1.32.0`
+- ModuleTemplate `kyma-system/telemetry-fast` pointing to `1.34.0`
+- ModuleTemplate `kyma-system/telemetry-experimental` pointing to `1.34.0-experimental` (only in DEV and STAGE)
+- ModuleTemplate `kyma-system/telemetry-dev` pointing to `1.35.0-rc1` (only in DEV)
+- ModuleReleaseMeta `kyma-system/telemetry`
+
+#### Old ModuleTemplates
+- ModuleTemplate `kyma-system/telemetry-1.32.0`
+- ModuleTemplate `kyma-system/telemetry-1.34.0`
+- ModuleTemplate `kyma-system/telemetry-1.34.0-experimental` (only in DEV and STAGE)
+- ModuleTemplate `kyma-system/telemetry-1.35.0-rc1` (only in DEV)
+
+As the new module metadata takes precedence, the reconciliation of the module already happens based on the new metadata. Since all versions and channel mappings are the same, no update is performed and all modules stay in the same state as before.
+
+The functionality can further be verified by enabling the module in a test SKR which will install it from scratch using the new metadata.
+
+### 4) [FAILURE] Rollback the New Module Metadata
```

Comment:
It looks to me that troubleshooting starts at this point. Can we create a section for the steps below? 10 steps for one procedure are already a lot, chunking can be a good idea.